driver/config/config.go

```diff
@@ -561,13 +562,29 @@ func (p *Config) CourierTemplatesRoot() string {
 	return p.p.StringF(ViperKeyCourierTemplatesPath, "/courier/template/templates")
 }
 
+func splitUrlAndFragment(s string) (string, string) {
+	i := strings.IndexByte(s, '#')
+	if i < 0 {
+		return s, ""
+	}
+	return s[:i], s[i+1:]
+}
+
 func (p *Config) parseURIOrFail(key string) *url.URL {
-	u, err := url.ParseRequestURI(p.p.String(key))
+	u, frag := splitUrlAndFragment(p.p.String(key))
+	url, err := url.ParseRequestURI(u)
 	if err != nil {
 		p.l.WithError(errors.WithStack(err)).
 			Fatalf("Configuration value from key %s is not a valid URL: %s", key, p.p.String(key))
 	}
-	return u
+	if url.Host == "" || (url.Scheme != "http" && url.Scheme != "https") {
```

Comment:
We allow other schemes as well (which is why tests fail) - e.g. `base64` or `file`

Reply:
I am a bit confused. [There ](https://github.com/ory/kratos/pull/1040#discussion_r571103306) you said that we should check for the scheme (`http(s)://`). So should I remove the scheme check? Sorry for the misunderstanding.

Reply:
No my bad, I was just not careful enough in my comment. `url.Scheme == ""` should be enough!